src/backends/tensorrt/plan_backend.cc

```diff
@@ -2780,11 +2799,15 @@ PlanBackend::Context::Run(
     }
   }
 
+  // Wait for the inference to be completed before copying output as output
+  // copy is on a separate stream
+  cudaEventSynchronize(events_[next_set_].ready_for_output_);
```

Comment:
Use `cudaStreamWaitEvent()` instead so the designated stream is waiting for the event not the host thread, see line 2718 for example